### ISSUE TITLE
'nb_epoch' affect the accuracy in the 1st epoch

### ISSUE THREAD

**Issue by chaiyujin** [fchollet/keras#2248]
```
train_set_x, train_set_y, valid_set_x, valid_set_y, test_set_x, test_set_y = load_data()

model = Sequential()
# input: 6  64x64 images  -> (6, 64, 64) tensors.
# this applies 3 convolution filters of size 5x5, 3x3, 3x3
model.add(Convolution2D(37, 5, 5, input_shape = (6, 64, 64), activation='relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
# 64 - 5 + 1 = 60 / 2 = 30

model.add(Convolution2D(256, 3, 3, activation='relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))
# 30 - 3 + 1 = 28 / 2 = 14

model.add(Convolution2D(800, 3, 3, border_mode='valid', activation='relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))
# 14 - 3 + 1 = 12 / 2 = 6

model.add(Flatten())
model.add(Dense(1200, activation = 'relu'))
model.add(Dropout(0.5))

model.add(Dense(82, activation = 'sigmoid'))

sgd = SGD(lr = 0.1, decay = 1e-6, momentum=0.9, nesterov = True)
model.compile(optimizer = sgd, loss = 'binary_crossentropy')

model.fit(train_set_x, train_set_y, batch_size = 400, nb_epoch = 30, 
          validation_data = (valid_set_x, valid_set_y), show_accuracy=True)
```

When `nb_epoch` is 30,  I got `loss: 1.8462 - acc:0.3767` on Epoch 1/30
When `nb_epoch` is 150, I got `loss: 1.5073 - acc:0.1432` on Epoch 1/150.

Is this normal?

**Comment by tboquet** [comment]
Do you fix the random seed?
```python
import numpy as np
np.random.seed(1337)
```

**Comment by chaiyujin** [comment]
@tboquet    Thanks, it works and  how the random seed affect the result?

**Comment by tboquet** [comment]
Your data is being shuffled during training. Fixing the random seed, you have the same sequence of mini-batches every time you run the code.

**Comment by chaiyujin** [comment]
@tboquet  Thank you so much.
